File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/dto/SysUserCreateRequest.java
package com.vipamp.vipclaw.admin.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

/**
 * 用户创建请求 DTO
 *
 * @author vipamp
 * @since 2026-03-05
 */
@Data
@Schema(description = "用户创建请求对象")
public class SysUserCreateRequest {

    /**
     * 用户名
     */
    @Schema(description = "用户名", example = "zhangsan", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "用户名不能为空")
    @Size(min = 3, max = 50, message = "用户名长度必须在 3-50 之间")
    @Pattern(regexp = "^[a-zA-Z0-9_]+$", message = "用户名只能包含字母、数字和下划线")
    private String username;

    /**
     * 密码
     */
    @Schema(description = "密码", example = "123456", requiredMode = Schema.RequiredMode.REQUIRED)
    @NotBlank(message = "密码不能为空")
    @Size(min = 6, max = 100, message = "密码长度必须在 6-100 之间")
    private String password;

    /**
     * 昵称
     */
    @Schema(description = "昵称", example = "张三")
    private String nickname;

    /**
     * 邮箱
     */
    @Schema(description = "邮箱", example = "zhangsan@example.com")
    @Email(message = "邮箱格式不正确")
    private String email;

    /**
     * 手机号
     */
    @Schema(description = "手机号", example = "13800138000")
    @Pattern(regexp = "^1[3-9]\\d{9}$", message = "手机号格式不正确")
    private String phone;

    /**
     * 性别 (0:女 1:男 2:未知)
     */
    @Schema(description = "性别 (0:女 1:男 2:未知)", example = "2")
    private Integer gender;

    /**
     * 头像 URL
     */
    @Schema(description = "头像 URL", example = "https://example.com/avatar.jpg")
    private String avatar;

    /**
     * 状态 (0:禁用 1:正常)
     **/
    @Schema(description = "状态 (0:禁用 1:正常)", example = "1")
    private Integer status;

    /**
     * 是否是管理员（0:否，1:是）
     */
    @Schema(description = "是否是管理员（0:否，1:是）", example = "0")
    private Integer isAdmin;
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/dto/SysUserCreateRequest.kt
package com.vipamp.vipclaw.admin.dto

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size

/**
 * User creation request object
 *
 * According to PRD document 3.2.1:
 * - Required fields: username, password, nickname
 * - Conditional required: email, phone (required for enterprise and public editions, optional for personal edition)
 * - Optional fields: gender, avatar
 * - Excluded: status, isAdmin (automatically set by system with default values)
 */
@Schema(description = "User creation request object")
data class SysUserCreateRequest(
    @Schema(description = "Username", example = "zhangsan", requiredMode = Schema.RequiredMode.REQUIRED)
    @Pattern(regexp = "^[a-zA-Z0-9_]+$", message = "Username can only contain letters, numbers and underscores")
    val username: String,

    @Schema(description = "Password (plaintext)", requiredMode = Schema.RequiredMode.REQUIRED)
    @Size(min = 6, max = 100, message = "Password length must be between 6-100")
    val password: String,

    @Schema(description = "Nickname", example = "John Doe", requiredMode = Schema.RequiredMode.REQUIRED)
    @Size(min = 1, max = 50, message = "Nickname length must be between 1-50")
    val nickname: String,

    @Schema(description = "Email", example = "zhangsan@example.com")
    val email: String? = null,

    @Schema(description = "Phone number", example = "13800138000")
    val phone: String? = null,

    @Schema(description = "Gender (0:female 1:male 2:unknown)", example = "2")
    val gender: Int? = 2,

    @Schema(description = "Avatar URL", example = "https://example.com/avatar.jpg")
    val avatar: String? = null,
)
